src/main/java/uk/gov/hmcts/darts/usermanagement/service/impl/UserManagementServiceImpl.java

```diff
@@ -168,6 +200,18 @@ private void updateEntity(UserPatch userPatch, UserAccountEntity userAccountEnti
 
         userAccountEntity.setLastModifiedBy(authorisationApi.getCurrentUser());
         userAccountEntity.setLastModifiedDateTime(OffsetDateTime.now());
+
+        return rolledBackTranscriptionsList;
+    }
+
+    private void unassignUserFromGroupsTheyArePartOf(UserAccountEntity entity) {
+        Set<SecurityGroupEntity> groupEntities = entity.getSecurityGroupEntities();
+        Iterator<SecurityGroupEntity> iterator = groupEntities.iterator();
+        while (iterator.hasNext()) {
+            SecurityGroupEntity groupEntity = iterator.next();
+            groupEntity.getUsers().remove(entity);
+            securityGroupRepository.save(groupEntity);
+        }
```

Comment:
we could simplify using a for loop:
```
for (SecurityGroupEntity groupEntity : groupEntities) {
            groupEntity.getUsers().remove(entity);
            securityGroupRepository.save(groupEntity);
        }
```

Reply:
I also wonder whether it would be good to clear the other side of the association, i.e. the collection `UserAccountEntity.securityGroupEntities`.